docs/app/routes/docs/guides/react-three-fiber.mdx

```diff
@@ -0,0 +1,401 @@
+---
+meta:
+  title: R3F | React Spring
+  'og:title': R3F | React Spring
+  'twitter:title': R3F | React Spring
+  description: A detailed guide to using React Spring with R3F.
+  'og:description': A detailed guide to using React Spring with R3F.
+  'twitter:description': A detailed guide to using React Spring with R3F.
+  'og:url': https://beta.react-spring.dev/docs/guides/react-three-fiber
+  'twitter:url': https://beta.react-spring.dev/docs/guides/react-three-fiber
+sidebar_position: 1
+---
+
+# React Three Fiber
+
+:::note
+It's assumed you have a base understanding of `react-spring` and `react-three-fiber`. If you're new to
+either, check out our [getting started](/docs/getting-started) or alternatively, the [r3f](https://docs.pmnd.rs/react-three-fiber) docs.
+:::
+
+## Introduction
+
+In this guide we'll explore why `react-spring` is a valuable addition to your `r3f` project, working with the [`imperative API`](/docs/concepts/imperative-api)
+to create performant animation updates on objects in the scene graph and with the [event system](/docs/advanced/events) of the library
+to update parts of your scene that are not wrapped in our [`animated` HOC](/docs/concepts/animated-elements). To work with `react-spring`
+and `react-three-fiber` you'll need to install the `@react-spring/three` package.
+
+```bash
+yarn add @react-spring/three
+```
+
+## Why use the library?
+
+A common question asked is why use `react-spring` with `react-three-fiber` when you can use the `useFrame` hook to update your meshes & objects
+instead without knowing another API. Well this is a great question, lets consider a simple use-case.
+
+I have a distortion blob and I want to it to change color on click. You could do this with `useFrame` to perform frame by frame updates, `useRef`
+to access the material object and use the `THREE.Color.lerp` function, slowly incrementing by lerp value until the color is reached. But this then
+requires that I keep track of a `THREE` instance of `Color` which _can_ lead to memory leaks if not handled correctly and not only that, but it would
+be a lot of code. Then you'd need to think about writing your own easing functions. With `react-spring` you can do this in a few lines of code.
```

Comment:
I think it's important to mention that react-spring's animations are physically correct. It is critical of motion design for animations to look realistic; otherwise, it's fake and awkward -- look at Dribbble to see what I mean. That is not to say that linear easings don't have their place; they are more appropriate for reactive elements or interactions where a response must be immediate, but that is inbuilt into CSS and `Vector3#lerp`. Furthermore, the flexibility to start/stop and replay animations, particularly with state and device motion preferences, makes this a uniquely accessible library from both a DX and UX perspective. These are points I'd highlight upfront since no other library has come close to offering the above.

Reply:
Just want to second, true spring motion is a huge plus. Even damping is an approximation. The ability to design physically accurate, fully interruptible animations, especially for 3D user interactions, is necessary for a polished feel in an app. This video might give you some inspiration, the visuals are great: https://youtu.be/KPoeNZZ6H4s?t=195

Reply:
Another bonus that came to mind is framerate independence. You can get the same with damping functions, but then damping is much harder to design with since it has fixed dynamics (since all damping is doing is approximating spring motion).

Reply:
fantastic points 🙇🏼‍♀️